src/access/Subjects.h

```diff
@@ -0,0 +1,208 @@
+/*
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <limits>
+
+#include <access/AuthMode.h>
+#include <lib/core/DataModelTypes.h>
+#include <lib/core/InPlace.h>
+#include <lib/core/NodeId.h>
+#include <lib/support/Variant.h>
+
+namespace chip {
+namespace Access {
+
+struct PaseSubject
+{
+    static constexpr uint16_t kPasscodeId = 0;
```

Comment:
This is not used except in logging, right?  Why do we need this at all?

Reply:
This field follows chip spec.

`
Subject ID SHALL be of type uint64 with semantics depending on the entry’s AuthMode as follows:

PASE: Lower 16-bits → Passcode ID Upper 48-bits → all bits clear
`

And also in spec:

`
Note that any Passcode ID other than 0, which is the default commissioning passcode, is reserved for future use.
`

So `kPasscodeId` is a const of 0 now.

Reply:
This does not answer my question.  What is this field actually used for?  If it's not used for anything, why is it there?

Reply:
It is used to construct the `SubjectId`, which is well defined in the spec.

The `SubjectId` is not only used for logging, it can be used to fill `CaseAdminNode` field in `AddNOC` commands (maybe in the future PR).

Although  `SubjectId` for PASE subject is not being used to encode messages, it doesn't prevent us from building a `SubjectId` for PASE subject. The `GetSubjectId` function is fully aligned with spec.

Reply:
Used to construct the SubjectId from what?

> The SubjectId is not only used for logging, it can be used to fill CaseAdminNode field in AddNOC commands

That value is just a NodeId.  It does not need any of the Subject machinery, and I can't see any situation where it would.

> The GetSubjectId function is fully aligned with spec.

No, it's just not.  But if you think it is, please walk me through step by step exactly how it's aligned.